src/mbgl/shaders/gl/shader_info.cpp

```diff
@@ -0,0 +1,178 @@
+#include <mbgl/shaders/gl/shader_info.hpp>
+
+#include <mbgl/shaders/ubo_max_count.hpp>
+
+namespace mbgl {
+namespace shaders {
+
+UniformBlockInfo::UniformBlockInfo(std::string_view name_, std::size_t id_)
+    : name(name_),
+      id(id_),
+      binding(id_) {}
+
+const std::vector<UniformBlockInfo> ShaderInfo<BuiltIn::BackgroundShader, gfx::Backend::Type::OpenGL>::uniformBlocks = {
```

Comment:
For now vectors are fine here, once I can come back to the code generation PR these can be made into `std::array`s.

Reply:
I used vector here, because I am passing it here are param:
        ```
shader = ShaderProgramGL::create(glContext,
                                         programParameters,
                                         shaderName,
                                         firstAttribName,
                                         shaders::ShaderInfo<ShaderID, gfx::Backend::Type::OpenGL>::uniformBlocks,
                                         vert,
                                         frag,
                                         additionalDefines);
```

Reply:
We can adjust it later and for now it's fine like this or should I find a way an change it now to arrays?

Reply:
It's fine for now, I don't think there will be much difference moving to arrays. Just a thought that code generation can make a bit more maintainable.